blocks/api/registration.js

```diff
@@ -68,6 +68,12 @@ export function registerBlockType( name, settings ) {
 		);
 		return;
 	}
+	if ( 'focusable' in settings && ! ( isFunction( settings.focusable ) || isBoolean( settings.focusable ) ) ) {
```

Comment:
I've thought about this and I think I will change the implementation to do the following:

- Initially assume that no child components are focusable and set the state variable focusable to false
- Render the block and set the tabIndex of the border based on the focusable state
- After the block has rendered then calculate if any of the components within the block can be focused according to the rules here: https://stackoverflow.com/questions/1599660/which-html-elements-can-receive-focus
- If the block border is focused and a child element is focusable than transfer focus to the first focusable child
- Update the state variable focusable